src/extension/tools/node/runTaskTool.tsx

```diff
@@ -54,8 +54,8 @@ class RunTaskTool implements vscode.LanguageModelTool<IRunTaskToolInput> {
 		const taskRunDurationMs = taskEndTime - taskStartTime;
 		let totalDurationMs: number | undefined;
 
-		// Start with 500 to ensure the buffer has content
-		const checkIntervals = [1000, 100, 100, 100, 100, 100];
+		// Start with 1000 to ensure the buffer has content, wait up to 20 seconds for the task to become idle
+		const checkIntervals = [1000, 100, 100, 100, 100, 100, 200, 200, 200, 200, 200, 200, 500, 500, 500, 1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000];
```

Comment:
Easier to understand approach would be to have target time (`20000`) and then just double the interval up to a maximum. Something like this:

```ts
const endTime = Date.now() + 20000;
let currentInterval = 500;
while (Date.now() < endTime) {
  currentInterval = Math.min(currentInterval * 2, 2000);
  await timeout(currentInterval);
}
```

That also makes it wait close to 20s, as opposed to 20 seconds plus all the request durations combined.